src/ActionBuilder/ProductType/ChangeOrderOfAttributes.php

```diff
@@ -43,16 +43,23 @@ public function createUpdateActions(
         $oldOrder = array_column($oldData['attributes'], 'name');
         $newOrder = array_column($sourceObject->getAttributes()->toArray(), 'name');
 
-        $hasNewValues = count(array_diff($newOrder, $oldOrder)) > 0;
-        $hasDeletedValues = count(array_diff($oldOrder, $newOrder)) > 0;
-
         // If amount and values same but not the order
-        if (!$hasNewValues && !$hasDeletedValues && $oldOrder !== $newOrder) {
+        if ($oldOrder !== $newOrder) {
             return [new ProductTypeChangeAttributeOrderByNameAction([
                 'attributeNames' => $newOrder
             ])];
         }
 
         return [];
     }
+
+    /**
+     * Must be the last action in our request (so all added and added attributes are done)
```

Comment:
> so all **added** and **added** attributes are done
I think you meant "all **added** and **changed** ..."

Reply:
"all added and removed", because this directly affects the order of the attributes. (maybe should a new added attribute an top or something like that)